### PR TITLE
G/f/g4asfunction/src/tier ii gmsh

### PR DESCRIPTION
**Pull request by zyang2022-5**
Created code for generating Tile that is axisymmetric to x=0 and z=0.
Created command line arguments -nX and -nodesec to control the number of nodes in X and Z respectively.

### CODE REVIEW

File: Makefile.localgmsh
Comment (grealesguti) [inline Makefile.localgmsh:1]: To accept the pull request I need you to create a new branch where you introduce the changes that you need locally and you only submit the changes for the main TierII branch!! You can do this with cherry-pick, or a little more lazy, not commiting the changes to Makefile and sim.
Example of cherry-pick to not merge some commits: 
` git rev-list --reverse fromgerritSHA1..togerritSHA1 | grep -vE 'skiphash1|skiphash2' | git cherry-pick --stdin`

The code you've provided is a command-line script for Git, a version control system. This script is used to cherry-pick commits from a range of commits, excluding certain commits that you don't want to include. Here's a breakdown of what each part of the script does:

* git rev-list --reverse fromgerritSHA1..togerritSHA1: This command lists all the commit hashes in the range from fromgerritSHA1 to togerritSHA1. The --reverse option is used to list the hashes in the order they were committed (i.e., oldest to newest).
* grep -vE 'skiphash1|skiphash2': This command filters out the commit hashes that you don't want to include in the cherry-pick. The -v option inverts the matching, so it excludes lines that match the pattern. The -E option allows for extended regular expressions, and the 'skiphash1|skiphash2' pattern matches either skiphash1 or skiphash2.
* git cherry-pick --stdin: This command applies the changes introduced by the remaining commit hashes to the current branch. The --stdin option means that the commit hashes are read from the standard input (which is the output of the previous commands).
So, in summary, this script cherry-picks all commits in a range, excluding certain commits. You can add more commit hashes to exclude by adding them to the grep command, separated by |.

The other way is for me to accept this and then replace these files with a previos version which is not the 'cleanest' way, and certainly not something reviewers want to deal with all the time. In case that this is easier for you now you should be able to make a new branch and do:
`git checkout origin/<gitbranch>-- <filename>`
to overwrite these files, make a new commit and make the pull request. If you do the pull request after this for both files it should be ok for me to merge.

Notice as well that even in the first commit you already have changes to GmshLYSO.cc

File: sim.cc
Comment (grealesguti) [inline sim.cc:1]: This other file should also remain the same

File: src_G4/construction.cc
Comment (grealesguti) [inline src_G4/construction.cc:1]: Check the YposTol2, i thing this is a tolerance for the Y position of the SiPM in the bar configuration, it should be zero in the tile configuration and not doing anything here.

**Review by grealesguti — CHANGES_REQUESTED** [review]

File: Makefile.localgmsh
Comment (zyang2022-5) [reply Makefile.localgmsh:1]: Thanks for the explanation. I was aware of "cherry-pick" but as you said my commits have changes to multiple files because I forgot to setup a git remote before I started making changes and I forgot which commit corresponds to what (i know in theory each commit is suppose to be one thing but I didn't quite follow that at the beginning). I needed a separate file on my system to run the TierII stuff in singularity anyways so I just created a separate file with the TierII conditions and pushed that into a new branch called "geom_merge" and created another pull request. You should be abele to merge seamlessly with that one.

File: src_G4/construction.cc
Comment (zyang2022-5) [reply src_G4/construction.cc:1]: YposTol2 is indeed 0.
